### ISSUE TITLE
[libraw] unable to build libraw:x86-uwp

### ISSUE THREAD

**Issue by fredstoolbox** [microsoft/vcpkg#3755]
Trying to build libraw out of the current master trunk using [x86-uwp]

got the following errors: 

5>d:\data\vcpkg_libraw\vcpkg\buildtrees\libraw\src\libraw-0.18.2\internal\dcraw_common.cpp(87): error C4996: 'strnicmp': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _strnicmp. See online help for details. [D:\Data\vcpkg_libraw\vcpkg\buildtrees\libraw\x86-uwp-dbg\raw_r.vcxproj]
         c:\program files (x86)\windows kits\10\include\10.0.17134.0\ucrt\string.h(560): note: see declaration of 'strnicmp'
     5>d:\data\vcpkg_libraw\vcpkg\buildtrees\libraw\src\libraw-0.18.2\internal\dcraw_common.cpp(227): error C4996: 'swab': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _swab. See online help for details. [D:\Data\vcpkg_libraw\vcpkg\buildtrees\libraw\x86-uwp-dbg\raw_r.vcxproj]
         c:\program files (x86)\windows kits\10\include\10.0.17134.0\ucrt\stdlib.h(1338): note: see declaration of 'swab'
     5>d:\data\vcpkg_libraw\vcpkg\buildtrees\libraw\src\libraw-0.18.2\internal\dcraw_common.cpp(1434): error C4146: unary minus operator applied to unsigned type, result still unsigned [D:\Data\vcpkg_libraw\vcpkg\buildtrees\libraw\x86-uwp-dbg\raw_r.vcxproj]
     5>d:\data\vcpkg_libraw\vcpkg\buildtrees\libraw\src\libraw-0.18.2\src\libraw_datastream.cpp(251): error C4996: 'fileno': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _fileno. See online help for details. [D:\Data\vcpkg_libraw\vcpkg\buildtrees\libraw\x86-uwp-dbg\raw_r.vcxproj]
         c:\program files (x86)\windows kits\10\include\10.0.17134.0\ucrt\stdio.h(2459): note: see declaration of 'fileno'
     5>d:\data\vcpkg_libraw\vcpkg\buildtrees\libraw\src\libraw-0.18.2\src\libraw_datastream.cpp(627): error C4996: 'fileno': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _fileno. See online help for details. [D:\Data\vcpkg_libraw\vcpkg\buildtrees\libraw\x86-uwp-dbg\raw_r.vcxproj]
         c:\program files (x86)\windows kits\10\include\10.0.17134.0\ucrt\stdio.h(2459): note: see declaration of 'fileno'
     5>d:\data\vcpkg_libraw\vcpkg\buildtrees\libraw\src\libraw-0.18.2\src\libraw_cxx.cpp(1739): error C4996: 'stricmp': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _stricmp. See online help for details.d:\data\vcpkg_libraw\vcpkg\buildtrees\libraw\src\libraw-0.18.2\src\libraw_datastream.cpp(652): error C3861: 'CreateFile': identifier not found [D:\Data\vcpkg_libraw\vcpkg\buildtrees\libraw\x86-uwp-dbg\raw_r.vcxproj]

**Comment by fredstoolbox** [comment]
looks like libraw has not been migrate to uwp build. I was able to build x86-Windows with no issues.

In order to build uwp, I have changed:
 - CMakeFileList.txt:
LN 97: change to: IF (MSVC)
LN102, add: ADD_DEFINITIONS(-D_CRT_SECURE_NO_DEPRECATE -D_CRT_NONSTDC_NO_DEPRECATE /wd4146 /wd4703)

- libraw_datastream.cpp
LN 652, replace the old 'CreateFile(blah)' to: HANDLE hFile = CreateFile2(sFile, GENERIC_READ, 0, OPEN_EXISTING, 0);

It now builds arm-uwp and x86-uwp no problem. Haven't got time to test the library yet.

**Comment by fredstoolbox** [comment]
Just another update, the dll appears to work.

**Comment by KindDragon** [comment]
Can you provide pull request with this changes to libraw https://github.com/LibRaw/LibRaw?

**Comment by dingsheng999** [comment]
It will be really awesome to see Libraw really working for UWP rather than passing compilation. In UWP we have to use StorageFile rather than using fopen to read files. It appears to be a lot of work and someone dedicated to the UWP migration might be needed.

**Comment by JackBoosY** [comment]
Hi @fredstoolbox, when I built libraw with **uwp**, I encountered the following error:
- dcraw_common.cpp(147): error C4996: 'strnicmp': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _strnicmp. See online help for details. [F:\vcpkg\buildtrees\libraw\x86-uwp-dbg\raw_r.vcxproj]
- dcraw_common.cpp(331): error C4996: 'swab': The POSIX name for this item is deprecated
- libraw_datastream.cpp(674): error C4996: 'fileno': The POSIX name for this item is deprecated
- libraw_datastream.cpp(700): error C3861: 'CreateFile': identifier not found
- libraw_cxx.cpp(2014): error C4996: 'stricmp': The POSIX name for this item is deprecated.
- dcraw_common.cpp(1713): error C4146: unary minus operator applied to unsigned type, result still unsigned
- dcraw_common.cpp(331): error C4996: 'swab': The POSIX name for this item is deprecated.

It seems that libraw does **NOT** currently support uwp, we need to wait to upgrade its uwp support version.

Thanks.
